src/e2e-test/java/io/cdap/plugin/common/ValidationHelper.java

```diff
@@ -0,0 +1,278 @@
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+import java.util.Base64;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Validation Helper.
+ */
+public class ValidationHelper {
+  private static final String PROJECT_ID = PluginPropertyUtils.pluginProp("projectId");
+  private static final Gson gson = new Gson();
+
+  public static boolean validateBQDataToGCS(String table, String bucketName) throws IOException,
+    InterruptedException {
+    Map<String, JsonObject> expectedData = new HashMap<>();
+    Map<String, JsonObject> actualData = listBucketObjects(bucketName);
+
+    getBigQueryTableData(table, expectedData);
+
+    boolean isMatched = matchJsonMaps(expectedData, actualData);
+
+    if (isMatched) {
+      System.out.println("The data is matched.");
+    } else {
+      System.out.println("The data is not matched.");
+    }
+
+    return isMatched;
+  }
+
+  public static boolean validateGCSDataToBQ(String bucketName, String table) throws IOException,
+    InterruptedException {
+    Map<String, JsonObject> expectedData = new HashMap<>();
+    Map<String, JsonObject> actualData = listBucketObjects(bucketName);
+
+    getBigQueryTableData(table, expectedData);
+
+    boolean isMatched = matchJsonMaps(expectedData, actualData);
+
+    if (isMatched) {
+      System.out.println("The data is matched.");
+    } else {
+      System.out.println("The data is not matched.");
+    }
+
+    return isMatched;
+  }
+
+  public static Map<String, JsonObject> listBucketObjects(String bucketName) {
+    Map<String, JsonObject> dataMap = new HashMap<>();
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    Page<Blob> blobs = storage.list(bucketName);
+
+    List<Blob> bucketObjects = StreamSupport.stream(blobs.iterateAll().spliterator(), true)
+      .filter(blob -> blob.getSize() != 0)
+      .collect(Collectors.toList());
+
+    Stream<String> objectNamesWithData = bucketObjects.stream().map(blob -> blob.getName());
+    List<String> bucketObjectNames = objectNamesWithData.collect(Collectors.toList());
+
+    if (!bucketObjectNames.isEmpty()) {
+      String objectName = bucketObjectNames.get(0);
```

Comment:
Why are we hardcoding 0th index there can be more than 1 file. Or even if there is 1 file only it creates a bottle neck you will have to handle it.

Reply:
As of now we are using 1 file in all scenarios.